exposee-app/Routes/users.js

```diff
@@ -1,18 +1,102 @@
 import express from "express";
 import bcrypt from "bcrypt";
 import { User } from "../models/users.js";
+import { Gift } from "../models/gifts.js";
+import { Wallet } from "../models/wallet.js";
 import { Op } from "sequelize";
 import pkg from "jsonwebtoken";
 const { sign } = pkg;
 import validate_Token from "../authentoken.js";
+import dotenv from "dotenv";
 
 const router = express.Router();
-
+dotenv.config();
+function verifytoken(req, res, next) {
+  const token = req.header("Authorization");
+  if (!token) {
+    return res.status(401).json({ error: "Acess denied. No token provided" });
+  }
+  jwt.verify(token, secretKey, (err, decoded) => {
+    if (err) {
+      return res.status(401).json({ error: "invalid token" });
+    }
+    req.userId = decoded.userId;
+    next();
+  });
+}
 router.get("/user", validate_Token, async (req, res) => {
   const users = await User.findAll();
   console.log(users);
   res.status(200).json(users);
 });
+router.get('/wallet/:user_id', validate_Token, async (req, res) => {
+  try {
+    const user_id = req.userId;
+    // const user_id = req.params.user_id;
+    const wallet = await Wallet.findOne({ where: { user_id } });
+    if (!wallet) {
+      return res.status(404).json({ error: 'Wallet not found' });
+    }
+
+    return res.json({ balance: wallet.amount });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+});
+router.post('/gift', validate_Token, async (req, res) => {
+  try {
+    const { sender_id, receiver_id, amount , video_id } = req.body;
+
+    // Check if both sender and receiver exist in the database
+    const senderWallet = await Wallet.findOne({ where: { user_id: sender_id } });
```

Comment:
the sender here should actually come from the logged in session.

otherwise, logged in user "A" can edit the js post body such that he sends money from everyone's wallet to A's wallet